devito/operator.py

```diff
@@ -96,6 +96,17 @@ def expr_cse(expr):
                     s_dict[arg] = Indexed(to_revert[value.base.label], *value.indices)
         to_revert[temp] = value.xreplace(s_dict)
 
+    # Make sure that no temporary that has to be reverted depends
+    # on another temporary of the list
+    changing = True
+    while changing:
+        changing = False
+        for temp, value in to_revert.items():
+            n_value = value.xreplace(to_revert)
```

Comment:
That's weird. Since `temp41` and `temp35` are not kept you should get a compilation error. Can I see the C code?